tests/test_ell1h.py

```diff
@@ -12,111 +13,204 @@
+        log.debug("Runing derivative for %s", "d_delay_d_" + p)
+        ndf = modelJ1853.d_phase_d_param_num(toasJ1853, p, testp[p])
+        adf = modelJ1853.d_phase_d_param(toasJ1853, delay, p)
+        diff = adf - ndf
+        if not np.all(diff.value) == 0.0:
+            mean_der = (adf + ndf) / 2.0
+            relative_diff = np.abs(diff) / np.abs(mean_der)
+            # print "Diff Max is :", np.abs(diff).max()
+            msg = (
+                "Derivative test failed at d_delay_d_%s with max relative difference %lf"
+                % (p, np.nanmax(relative_diff).value)
+            )
+            if p in ["EPS1DOT", "EPS1"]:
+                tol = 0.05
             else:
-                continue
+                tol = 1e-3
+            log.debug(
+                "derivative relative diff for %s, %lf"
+                % ("d_delay_d_" + p, np.nanmax(relative_diff).value)
+            )
+            assert np.nanmax(relative_diff) < tol, msg
+        else:
+            continue
 
-    def test_J0613_H4(self):
-        log = logging.getLogger("TestJ0613.fit_tests")
-        f = ff.GLSFitter(self.toasJ0613, self.modelJ0613)
-        f.fit_toas()
-        f.model.free_params = ("H3", "H4")
-        for pn, p in f.model.get_params_dict("free", "quantity").items():
-            op = getattr(f.model_init, pn)
-            diff = np.abs(p.value - op.value)
-            sigma = diff / op.uncertainty_value
-            # Fit th
-            assert (
-                sigma < 0.7
-            ), "refit %s is %lf sigma different from original value" % (pn, sigma)
-
-    def test_J0613_STIG(self):
-        log = logging.getLogger("TestJ0613.fit_tests_stig")
-        f = ff.GLSFitter(self.toasJ0613, self.modelJ0613_STIG)
+
+def test_J0613_fit_with_H3_H4(toasJ0613, modelJ0613):
+    log = logging.getLogger("TestJ0613.fit_tests")
+    f = ff.GLSFitter(toasJ0613, modelJ0613)
+    f.fit_toas()
+
+
+def test_J0613_STIG(toasJ0613, modelJ0613_STIG):
+    log = logging.getLogger("TestJ0613.fit_tests_stig")
+    f = ff.GLSFitter(toasJ0613, modelJ0613_STIG)
+    f.fit_toas()
+
+
+def test_no_H3_H4(toasJ0613):
+    """ Test no H3 and H4 in model.
+    """
+    no_H3_H4 = simple_par.replace("H4 2.0262048E-7  1       1.1276173E-7", "")
+    no_H3_H4 = no_H3_H4.replace("H3 2.7507208E-7  1       1.5114416E-7", "")
+    print(no_H3_H4)
+    no_H3_H4_model = model.get_model(StringIO(no_H3_H4))
+    assert no_H3_H4_model.H3.value == None
+    assert no_H3_H4_model.H4.value == None
+    test_toas = toasJ0613[::20]
+    f = ff.WLSFitter(test_toas, no_H3_H4_model)
+    f.fit_toas()
+
+
+def test_H3_and_H4_non_zero(toasJ0613):
+    """ Testing if the different H3, H4 combination breaks the fitting. the
+        fitting result will not be checked here.
+    """
+    simple_model = model.get_model(StringIO(simple_par))
+
+    test_toas = toasJ0613[::20]
+    f = ff.WLSFitter(test_toas, simple_model)
+    f.fit_toas()
+
+
+def zero_H4(toasJ0613):
```

Comment:
This needs to be called `test_zero_H4`  for pytest to find it; you can use `pytest -v tests/test_ell1h.py` to have it print out the names of the tests it is actually running.

Reply:
Good catch.